File: homework-10-ui/src/dto/UpdateBookDto.ts

export class UpdateBookDto {
    title: string;
    year: string;
    authorId: number;
    genreId: number;

    constructor(title: string, year: string, authorId: number, genreId: number) {
        this.title = title;
        this.year = year;
        this.authorId = authorId;
        this.genreId = genreId;
    }
}

File: homework-11/src/main/java/ru/otus/homework/dto/book/UpdateBookDto.java
package ru.otus.homework.dto.book;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PastOrPresent;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.Year;

@Data
@AllArgsConstructor
public class UpdateBookDto {
    @NotBlank(message = "Title should'n be empty")
    private final String title;

    @NotNull(message = "Year should'n be empty")
    @PastOrPresent(message = "Year should present or past")
    private final Year year;

    @NotBlank(message = "Author should'n be empty")
    private final String authorId;

    @NotBlank(message = "Genre should'n be empty")
    private final String genreId;
}
